logback/src/main/java/com/grafana/loki/LokiAppender.java

```diff
@@ -0,0 +1,205 @@
+package com.grafana.loki;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.locks.ReentrantLock;
+
+import com.google.protobuf.Timestamp;
+import com.grafana.loki.entry.Entry;
+import com.grafana.loki.logproto.Logproto.Entry.Builder;
+
+import ch.qos.logback.classic.pattern.CallerDataConverter;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.classic.spi.ThrowableProxyUtil;
+import ch.qos.logback.core.UnsynchronizedAppenderBase;
+import ch.qos.logback.core.encoder.EchoEncoder;
+import ch.qos.logback.core.encoder.Encoder;
+
+public class LokiAppender<E> extends UnsynchronizedAppenderBase<E> {
```

Comment:
Do we want this or to extend `AppenderBase` I'm not sure if we go with Unsynchronized exactly what we are on the hook for handling in regards to thread safety

Reply:
We could go with Unsynchronized by using  `ReentrantLock` to make sure only one thread gets the access. However, let's just use `AppenderBase` to keep things simple.